testbins/test-hal-scb/src/main.rs

```diff
@@ -0,0 +1,100 @@
+#![no_std]
+#![no_main]
+
+extern crate aerugo;
+extern crate calldwell;
+extern crate cortex_m;
+extern crate cortex_m_rt as rt;
+
+use aerugo::hal::user_peripherals::{CPUID, SCB};
+use aerugo::{Aerugo, InitApi, SystemHardwareConfig};
+use calldwell::write_str;
+use rt::entry;
+
+// Align for dcache line size
+#[repr(align(32))]
+#[derive(Debug)]
+struct DummyData {
+    // D-cache line size is 32B, data must be multiple of 32B
+    // 128B spans 4 d-cache lines.
+    pub data: [u8; 128],
+}
+
+impl Default for DummyData {
+    fn default() -> Self {
+        Self { data: [0u8; 128] }
+    }
+}
+
+fn perform_dcache_tests(scb: &mut SCB, cpuid: &mut CPUID) {
+    // Check cache enabling/disabling
+    scb.enable_dcache(cpuid);
+    assert!(
+        SCB::dcache_enabled(),
+        "D-Cache was enabled, yet it's reported as disabled"
+    );
+
+    // Clearing and invalidating cache has no observable effect outside of processor's core.
+    // Exact behavior was verified by analysis.
+    let mut dummy = DummyData::default();
+    // Safety: This is safe as long as `dummy` is aligned to 32 bytes,
+    // and it's data length is a multiple of 32.
+    assert!(core::mem::align_of::<DummyData>() % 32 == 0);
+    assert!(core::mem::size_of::<DummyData>() % 32 == 0);
+    unsafe {
+        scb.clean_dcache_by_slice(&dummy.data);
+        scb.invalidate_dcache_by_slice(&mut dummy.data);
+    }
+
+    scb.clean_invalidate_dcache(cpuid);
+
+    // TODO: Why does this crash/halt the test?
```

Comment:
I think we know why now?

Reply:
i've changed this comment to describe our current issue better